Ports/sdcc/patches/0001-Recognize-Serenity-as-a-system.patch

```diff
@@ -0,0 +1,21 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?kleines=20Filmr=C3=B6llchen?= <filmroellchen@serenityos.org>
+Date: Sun, 21 Aug 2022 17:12:00 +0200
+Subject: [PATCH] Recognize Serenity as a system
+
+---
+ config.sub | 1 +
+ 1 file changed, 1 insertion(+)
+
+diff --git a/config.sub b/config.sub
+index 1d8e98b..6ed7f57 100755
+--- a/config.sub
++++ b/config.sub
+@@ -1390,6 +1390,7 @@ case $os in
+ 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
+ 	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* \
+ 	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -redox* | -bme* \
++			| -serenity* \
+ 	      | -midnightbsd*)
+ 	# Remember, each alternative MUST END IN *, to match a version number.
+ 		;;
```

Comment:
This should probably be replaced with `use_fresh_config_sub`/`config_sub_paths`.

Reply:
Again; the config script needs a host argument or it doesn't work!

Reply:
Yes? That's exactly what `use_fresh_config_sub` does. It downloads a copy of `config.sub` from the GNU git server (which has Serenity support already included) to replace the outdated copy that is shipped inside the tarball, no manual patching needed.

Reply:
ah okay, is that even documented? I couldn't find it in the port documentation.

Reply:
[Yes](https://github.com/SerenityOS/serenity/blob/master/Ports/README.md#use_fresh_config_sub), although it doesn't explicitly mention that this is a viable replacement for patching in Serenity support manually.